### ISSUE TITLE
[question] any way to connect code to "registerAction()"?

### ISSUE THREAD

**Issue by 1awless** [Ladysnake/Blabber#10]
i want to execute code when a player chooses a specific action in my dialogue.
![image](https://github.com/Ladysnake/Blabber/assets/98748787/ce6478f4-6609-4bb9-b761-9a672a0c7b33)

**Comment by Pyrofab** [comment]
Sorry for the delay, I had missed this issue. You need to pass either a full object or a [lambda expression](https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html) here - so in your case probably more something like `ExampleMod::doThisAction`.

**Comment by NotTheRealRoKur0** [comment]
sorry to bother again, but how do i use this actually?
![image](https://github.com/Ladysnake/Blabber/assets/98748787/eb0de8f1-ea50-4cb4-9038-80eea853e1d4)
it doesnt seem to work when i select the action.
i put the code in the main class of my mod (and the registerAction in the onInitialize function)

(i tried lambda too but it doesnt seem to work either)

**Comment by NotTheRealRoKur0** [comment]
filename firstjoin.json (if that matters)
[json.json](https://github.com/Ladysnake/Blabber/files/14277466/json.json)

i think ive tried using cca before and i didnt do well (kinda new to this stuff).
so ill just stick with this for now, but ill look into it sometime again!

**Comment by Pyrofab** [comment]
You don't have any action in that JSON, which is why your action won't be called. You need to add something like this to one of your states :
```json
   "chosen_their": {
      "action": {
        "type": "yourmod:chosen_their"
      }
```

**Comment by NotTheRealRoKur0** [comment]
```
net.minecraft.util.crash.CrashException: mouseClicked event handler
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen.java:414) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.Mouse.onMouseButton(Mouse.java:98) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.Mouse.method_22686(Mouse.java:169) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor.java:102) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.Mouse.method_22684(Mouse.java:169) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:build 7]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3474) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:237) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1244) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.MinecraftClient.run(MinecraftClient.java:802) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-merged-8c37870d34-1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.2-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.3.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) ~[dev-launch-injector-0.2.1+build.8.jar:?]
Caused by: java.lang.RuntimeException: Not a JSON object: null; Not a JSON object: null; Not a JSON object: null missed input: {"chosen_their":{"text":"","choices":[],"type":"end_dialogue","action":{"type":"example-mod:chosen_their"}},"chosen_theyre":{"text":"","choices":[],"type":"end_dialogue","action":{"type":"example-mod:chosen_theyre"}},"chosen_there":{"text":"","choices":[],"type":"end_dialogue","action":{"type":"example-mod:chosen_there"}}}
	at com.mojang.serialization.DataResult.lambda$getOrThrow$5(DataResult.java:101) ~[datafixerupper-6.0.8.jar:?]
	at com.mojang.datafixers.util.Either$Right.map(Either.java:99) ~[datafixerupper-6.0.8.jar:?]
	at com.mojang.serialization.DataResult.getOrThrow(DataResult.java:93) ~[datafixerupper-6.0.8.jar:?]
	at org.ladysnake.blabber.impl.common.DialogueLoader.lambda$load$2(DialogueLoader.java:74) ~[blabber-1.4.0-mc1.20.1.jar:?]
	at java.util.TreeMap.forEach(TreeMap.java:1282) ~[?:?]
	at org.ladysnake.blabber.impl.common.DialogueLoader.lambda$load$3(DialogueLoader.java:70) ~[blabber-1.4.0-mc1.20.1.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
```

with:
[json.json](https://github.com/Ladysnake/Blabber/files/14289703/json.json)
